src/compiler/emitter.ts

```diff
@@ -6891,12 +6891,13 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, Promi
                 if (currentSourceFile.text.charCodeAt(comment.pos + 1) === CharacterCodes.asterisk) {
                     return currentSourceFile.text.charCodeAt(comment.pos + 2) === CharacterCodes.exclamation;
                 }
+                // Omit /// comment if compilerOptions.removeComments is true.
                 // Verify this is /// comment, but do the regexp match only when we first can find /// in the comment text
                 // so that we don't end up computing comment string and doing match for all // comments
-                else if (currentSourceFile.text.charCodeAt(comment.pos + 1) === CharacterCodes.slash &&
+                if (!compilerOptions.removeComments && (currentSourceFile.text.charCodeAt(comment.pos + 1) === CharacterCodes.slash &&

```

Comment:
I agree with mohamed.  I think it should be something like "shoudlEmitComment" and it can take a boolean saying if we're before the first token or not.  Then:

1.  We always emit pinned comments.
2. If we're before the first comment then we always emit triple-slash comments.
3. If we're after the first comment and it is not pinned, then we emit based on hte value of removeComments.